tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/api/PostResourceIntTest.java

```diff
@@ -397,6 +399,164 @@ public void shouldAllowMMultipleOtherSpecialties() throws Exception {
+    aSubspecialty.setSpecialtyTypes(new HashSet<>(
+        Collections.singletonList(SpecialtyType.SUB_SPECIALTY)));
+    em.persist(aSubspecialty);
+
+    PostSpecialty subspecialtyPostSpecialty = createPostSpecialty(aSubspecialty,
+        PostSpecialtyType.SUB_SPECIALTY, post);
+    post.getSpecialties().stream().findFirst().ifPresent(ps -> primaryPostSpecialty.setId(ps.getId()));
+    updatedPost.setSpecialties(new HashSet<>(Arrays.asList(primaryPostSpecialty, subspecialtyPostSpecialty)));
+
+    PostDTO updatedPostDto = postMapper.postToPostDTO(updatedPost);
+
+    restPostMockMvc.perform(put("/api/posts")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(TestUtil.convertObjectToJsonBytes(updatedPostDto)))
+        .andExpect(status().isOk());
+
+    Post resultingUpdatedPost = postRepository.findById(post.getId()).orElse(null);
+    Set<PostSpecialty> updatedPostSpecialtySet = resultingUpdatedPost.getSpecialties();
+    List <PostSpecialty> addedSubspecialties = updatedPostSpecialtySet.stream()
+        .filter(ps -> ps.getPostSpecialtyType().equals(PostSpecialtyType.SUB_SPECIALTY))
+        .collect(Collectors.toList());
+
+    assertThat(addedSubspecialties.size()).isEqualTo(1);
+    assertThat(addedSubspecialties.get(0).getSpecialty()).isEqualTo(aSubspecialty);
+  }
+
+  @Test
+  @Transactional
+  public void shouldFailToCreatePostIfSubspecialtyIsNotASubspecialty() throws Exception {
+    int databaseSizeBeforeCreate = postRepository.findAll().size();
+
+    // Attempt to save a Post that has a specialty of specialtyType PLACEMENT as a subspecialty
+    // (i.e. link Post and a Specialty that is SpecialtyType.PLACEMENT in a PostSpecialty in
+    // a PostSpecialty of PostSpecialtyType.SUB_SPECIALTY)
+    // The update should fail.
+
+    Post post = createEntity();
+    post.setNationalPostNumber("NEW_NPN");
+    Specialty notASubspecialty = createSpecialty();
+    notASubspecialty.setSpecialtyTypes(new HashSet<>(
+        Collections.singletonList(SpecialtyType.PLACEMENT)));
+
+    Specialty persistedNonSubspecialty = specialtyRepository.save(notASubspecialty);
+
+    PostSpecialty postSpecialty = createPostSpecialty(persistedNonSubspecialty,
+        PostSpecialtyType.SUB_SPECIALTY, post);
+    post.setSpecialties(new HashSet<>(Collections.singletonList(postSpecialty)));
+    PostDTO postDto = postMapper.postToPostDTO(post);
+
+    restPostMockMvc.perform(post("/api/posts")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(TestUtil.convertObjectToJsonBytes(postDto)))
+        .andExpect(status().isBadRequest())
+        .andExpect(jsonPath("$.message").value("error.validation"))
+        .andExpect(jsonPath("$.fieldErrors[0].field").value("specialties"));
+
+    List<Post> postList = postRepository.findAll();
+    assertThat(postList).hasSize(databaseSizeBeforeCreate);
+  }
+
+  @Test
+  @Transactional
+  public void shouldAllowCreationOfPostIfSubspecialtyIsASubspecialty() throws Exception {
```

Comment:
Maybe rename to '...IsOfTypeSubspecialty' for clarity

Reply:
same as above